x-pack/plugins/lens/public/app_plugin/save_modal_container_helpers.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { LensAppServices } from './types';
+import type { LensEmbeddableInput } from '../embeddable';
+import { LENS_EMBEDDABLE_TYPE } from '../../common/constants';
+
+export const redirectToDashboard = ({
+  embeddableInput,
+  dashboardFeatureFlag,
+  dashboardId,
+  originatingApp,
+  getOriginatingPath,
+  stateTransfer,
+}: {
+  embeddableInput: LensEmbeddableInput;
+  dashboardId: string;
+  dashboardFeatureFlag: LensAppServices['dashboardFeatureFlag'];
+  originatingApp?: string;
+  getOriginatingPath?: (dashboardId: string) => string | undefined;
+  stateTransfer: LensAppServices['stateTransfer'];
+}) => {
+  if (!dashboardFeatureFlag.allowByValueEmbeddables) {
+    throw new Error('redirectToDashboard called with by-value embeddables disabled');
+  }
+
+  const state = {
+    input: embeddableInput,
+    type: LENS_EMBEDDABLE_TYPE,
+  };
+
+  const path =
+    getOriginatingPath?.(dashboardId) ??
+    (dashboardId === 'new' ? '#/create' : `#/view/${dashboardId}`);
+  const appId = originatingApp || 'dashboards';
```

Comment:
This is the fix. It was originatingApp ?? 'dashboards' causing the problematic redirect